tests/drivers/pwm/pwm_api/src/test_pwm.c

```diff
@@ -82,12 +82,17 @@
 #define UNIT_USECS	1
 #define UNIT_NSECS	2
 
+const struct device *get_pwm_device(void)
+{
+	return device_get_binding(PWM_DEV_NAME);
+}
+
 static int test_task(uint32_t port, uint32_t period, uint32_t pulse, uint8_t unit)
 {
 	TC_PRINT("[PWM]: %" PRIu8 ", [period]: %" PRIu32 ", [pulse]: %" PRIu32 "\n",
 		port, period, pulse);
 
-	const struct device *pwm_dev = device_get_binding(PWM_DEV_NAME);
+	const struct device *pwm_dev = get_pwm_device();
```

Comment:
This is just factoring out a common call, right?  I don't see it being necessary for userspace support.

Reply:
No. It is to allow fetching the `struct device` pointer in `main.c` for granting access to it from user space.

Reply:
Huh?  But the struct device pointer isn't stored in the main module; and if it was it would probably be in a variable that couldn't be read from userspace.

Did this actually *fail* if you left the `device_get_binding(PWM_DEV_NAME)` call in place?

This refactoring makes sense to allow the PWM device to be obtained from a module that might not know `PWM_DEV_NAME`, but I can't see that it has any relation to the test running in userspace.  As such its presence the commit that switches to userspace confuses me (and might confuse somebody else looking for an example of how to convert another existing test).

Reply:
No, it does not fail. I have introduced the new function `get_pwm_device()` to unify how the PWM test device is obtained in `main.c` and `test_pwm.c`.

Previously, `main.c` did not need to obtain a `struct device` pointer to the PWM test device, but it needs to do that now in order to grant access to that kernel object from user space. This is similar to how other test cases handle access granting in `test_main()`.